hpy/devel/__init__.py

```diff
@@ -54,15 +77,45 @@ def fix_distribution(self, dist):
 
             Used from both setup.py and hpy/test.
         """
+        # ============= Distribution ==========
         dist.hpydevel = self
 
-        base_build = dist.cmdclass.get("build", build)
-        base_build_ext = dist.cmdclass.get("build_ext", build_ext)
-        orig_bdist_egg_write_stub = bdist_egg_mod.write_stub
+        @monkeypatch(dist.__class__)
+        def has_ext_modules(self):
+            if self.ext_modules or self.hpy_ext_modules:
+                return True
+            return False
 
-        if isinstance(base_build_ext, type):
+        # ============= build ==========
+        if cmd.build is not None:
+            build = dist.cmdclass.get("build", cmd.build.build)
+            build_hpy = make_mixin(build, build_hpy_mixin)
+            dist.cmdclass['build'] = build_hpy
+
+        # ============= build_ext ==========
+        build_ext = dist.cmdclass.get("build_ext", cmd.build_ext.build_ext)
+        self.build_ext_sanity_check(build_ext)
+        build_ext_hpy = make_mixin(build_ext, build_ext_hpy_mixin)
+        dist.cmdclass['build_ext'] = build_ext_hpy
+
+        # ============= bdist_egg ==========
+        @monkeypatch(setuptools.command.bdist_egg)
+        def write_stub(resource, pyfile):
+            """
+            This is needed because the default bdist_egg unconditionally writes a .py
+            stub, thus overwriting the one which was created by
+            build_ext_hpy_mixin.write_stub.
+            """
+            if resource.endswith(".hpy.so"):
+                log.info("stub file already created for %s", resource)
+                return
+            write_stub.super(resource, pyfile)
```

Comment:
When building NumPy/HPy and doing `rm -rf build` before `python3 setup.py --hpy-abi=universal  build_ext -if` (or when building the first time), then it fails with:

```
error: [Errno 2] No such file or directory: 'build/lib.macosx-12.3-x86_64-3.8-pydebug/numpy/core/_multiarray_umath.py'
```

This is because the parent dir `build/lib.macosx-12.3-x86_64-3.8-pydebug/numpy/core` does not yet exist.
Could we maybe create the parent dir of the stub here (if it does not exist)?

Reply:
I don't understand. This is a new bug, or an existing one?